script/tool/lib/src/native_test_command.dart

```diff
@@ -442,10 +468,11 @@ this command.
           .whereType<File>()
           .where(isTestBinary)
           .where((File file) {
-        // Only run the debug build of the unit tests, to avoid running the
-        // same tests multiple times.
+        // Only run the release build of the unit tests, to avoid running the
+        // same tests multiple times. Release is used rather than debug since
+        // `build-examples` builds release versions.
```

Comment:
Was going to say perhaps we should be changing to generate a debug build of build-examples given that it's typically more convenient to debug debug builds. That said, in the end you want to verify the release build of the library code in the plugin, and people shouldn't be relying on `#ifdef`'ed code in their tests, so this seems fine.

If this ever gets problematic, we could add a buildmode flag.

? Debugging debug builds of unit tests is typically more convenient, though

Reply:
I just realized that there's actually a more annoying problem, which is that the current behavior isn't going to be good for local use. Filed https://github.com/flutter/flutter/issues/89303.

I'm going to do that as a follow-up since it applies to the already-landed Windows flow in the same way, but I'll do it shortly.